samples/android/Continote/app/src/main/java/com/firebasecontinue/sample/continote/EditNoteActivity.java

```
@@ -0,0 +1,134 @@
+    @Nullable
+    private String mDatabaseKey = null;
+
+    // Firebase Realtime Database reference for the current user's Note to edit, based on the key.
+    @Nullable
+    private DatabaseReference mDatabaseRef = null;
+
+    // UI elements
+    @Nullable
+    private Button mContinueWritingElsewhereButton = null;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        Intent intent = getIntent();
+        mDatabaseKey = (intent != null) ? intent.getStringExtra("databaseKey") : null;
+        if (mDatabaseKey == null) {
+            // We must have a database key to be on this screen.
+            finish();
+            return;
+        }
+
+        setContentView(R.layout.activity_edit_note);
+
+        // Gather the UI elements for this Activity for future manipulation.
+        mContinueWritingElsewhereButton =
+                (Button) findViewById(R.id.continueWritingElsewhereButton);
+    }
+
+    @Override
+    protected void handleUserSignedIn(@NonNull FirebaseUser user) {
+        super.handleUserSignedIn(user);
+
+        if (user == null) {
+            // This should never happen, but just in case.
+            return;
+        }
+
+        // TODO
+    }
+
+    @Override
+    protected void handleUserSignedOut() {
+        super.handleUserSignedOut();
+
+        // The user must be signed in to view this screen, so navigate away from it if the user is
+        // signed out.
+        finish();
+    }
+
+    /**
+     * Handles when the user taps the continueWritingElsewhereButton.
+     *
+     * @param v The View that called this triggered this handler.
+     *          This should only be the continueWritingElsewhereButton itself.
+     */
+    public void handleContinueWritingElsewhereButtonTapped(@Nullable View v) {
+        if (!currentUserIsSignedIn() || mDatabaseKey == null) {
+            // This should never happen, but just in case.
```

I'd throw an AssertionError here instead of returning

---

Done (see comment above).